tests/Unit/Evolution/Systems/Ccz4/Test_Ricci.cpp

```diff
@@ -89,19 +91,24 @@ void test_compute_spatial_ricci_tensor(
   const auto inverse_conformal_spatial_metric =
       determinant_and_inverse(conformal_spatial_metric).second;
 
-  using conformal_spatial_metric_tag =
-      Ccz4::Tags::ConformalMetric<SpatialDim, Frame::Inertial, DataVector>;
-  Variables<tmpl::list<conformal_spatial_metric_tag>>
-      conformal_spatial_metric_var(num_points_3d);
-  get<conformal_spatial_metric_tag>(conformal_spatial_metric_var) =
-      conformal_spatial_metric;
-  const auto d_conformal_spatial_metric_var =
-      partial_derivatives<tmpl::list<conformal_spatial_metric_tag>>(
-          conformal_spatial_metric_var, mesh,
-          coord_map.inv_jacobian(x_logical));
-  const auto& d_conformal_spatial_metric =
-      get<Tags::deriv<conformal_spatial_metric_tag, tmpl::size_t<SpatialDim>,
-                      Frame::Inertial>>(d_conformal_spatial_metric_var);
+  gr::Solutions::KerrSchild::IntermediateVars<DataVector, Frame::Inertial>
```

Comment:
Can you do `const auto d_det.. = get<gr::Tags::DerivDetSpatialMetric<...>>(solution.variables(x, t, tmpl::list<gr::Tags::DerivDetSpatialMetric<...>>{}));` instead of relying on the intermediate vars?

Reply:
I could be wrong, but to be able to do this, wouldn't `DerivDetSpatialMetric` need to be included in [this list](https://github.com/sxs-collaboration/spectre/blob/fc92aea9d9c6403fdf0aab2d4859f76c95d8258b/src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Solutions.hpp#L29) that is common to all GR solutions? I didn't add it there when I added it to KerrSchild [here](https://github.com/sxs-collaboration/spectre/pull/3610).

Reply:
ah... I'm not sure it needs to be in the last, I didn't write the KS solution. However, I think the tag should be retrievable from `variables`. The `IntermediateVars` is really an implementation detail, not the way you should get variables from the solution...

Reply:
If I confirm that it does, do you feel it's appropriate to add this tag as something that is common to all of the Analytic GR solutions?

Reply:
It might be useful, but I don't think it should be necessary to add everywhere. That would be a rather restrictive design and we should figure out why it seems to be needed everywhere.

Reply:
Sorry, I jumped the gun with that question. I can confirm whether or not this is the case first, and then let you know

Reply:
Yes, this does appear to be the case. [This static assert](https://github.com/sxs-collaboration/spectre/blob/fc92aea9d9c6403fdf0aab2d4859f76c95d8258b/src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp#L256) seems to be failing because `DerivDetSpatialMetric` is not a tag in `tags<DataType, Frame>`, which is defined [here](https://github.com/sxs-collaboration/spectre/blob/fc92aea9d9c6403fdf0aab2d4859f76c95d8258b/src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Solutions.hpp#L29), which seems to be a list common to all GR solutions.

Let me know how you would like to proceed! Not sure if this is a design thing you want me or someone else to change, or if you want me to just compute it with the intermediate for now.

Reply:
does it work if you change the assert to use `tmpl::push_back<tags<DataType, Frame>, DerivDetSpatialMetric>` (the extra tag)?

Reply:
Yes, that does work. It makes me wonder if there was a reason that when this `variables` function was written, there was a reason to not allow intermediates to be retrieved in this way

Reply:
You don't need access to most of the intermediate quantities. In general you only need access to the 3+1 quantities. You don't need access to H, l_a, and all the other intermediate things.

Reply:
Ah yes. Did you want me to make the change to the static assert that you suggested? In light of that intermediates aren't things you typically want to retrieve, I had thought you were just asking me to do that to confirm that that was the problem. Let me know if this is a change you'd like me to make as part of this PR.

Reply:
Yes, please make the change to the assert. Sorry for not being clear on that!

Reply:
Also my bad! Will push the fix soon